src/Core/Transmission.BlobStorage/BlobStorageTransmissionSender.cs

```diff
@@ -28,24 +28,21 @@ public BlobStorageTransmissionSender(IBlobContainer container)
 
         /// <inheritdoc/>
         public async Task SendAsync(
-            IReadOnlyCollection<AttachmentDescriptor> batch,
+            IAsyncEnumerable<AttachmentDescriptor> batch,
             CancellationToken cancellationToken)
         {
             if (batch == null)
             {
                 throw new ArgumentNullException(nameof(batch));
             }
 
-            if (batch.Count > 0)
+            await foreach (AttachmentDescriptor attachment in batch
+                .WithCancellation(cancellationToken))
             {
                 try
                 {
-                    foreach (AttachmentDescriptor descriptor in batch)
-                    {
-                        await _container
-                            .UploadAsync(descriptor, cancellationToken)
-                            .ConfigureAwait(false);
-                    }
+                    await _container
```

Comment:
we should.have to upload conditions in order to batch count time ... like we try to upload 10 items but if we cannot get 10 items in a fixed amount of time we upload what we have.

Reply:
we could do it in next PR